### PR TITLE
Dockerise our ModelScan application

### PR DESCRIPTION
**Pull request by PE39806**
Write a Dockerfile for the REST Model Scan application.

The Dockerfile should expose the relevant ports.

The Image should be added to the dev docker compose build.

### CODE REVIEW

**Review by JR40159 — CHANGES_REQUESTED** [review]

**Review by JR40159 — CHANGES_REQUESTED** [review]
It might be worth fixing the example return objects in the Swagger docs

File: lib/modelscan_api/Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM python:3.12-slim
```
[the remark this answers is not in the document]
Comment (PE39806) [reply lib/modelscan_api/Dockerfile.dev:1]: I had not. It looks like `3.12-slim` resolves to `3.12-slim-bookend` which is one major Debian release ahead of `bullseye`. I'm happy to downgrade to `3.12-slim-bullseye` as there are no meaningful functionality changes.
Is there a reason we don't use the `slim` docker image for the docs build? If there is a known issue then I'd like to investigate that here too.

File: lib/modelscan_api/bailo_modelscan_api/main.py [hunk 2]
```diff
@@ -101,6 +130,7 @@ def scan_file(in_file: UploadFile, background_tasks: BackgroundTasks) -> dict[st
     finally:
         try:
             # Clean up the downloaded file as a background task to allow returning sooner.
+            # If using a temporary dir then this would happen anyway, but if Settings' download_dir evaluates then this is required.
```
Comment (JR40159) [inline lib/modelscan_api/bailo_modelscan_api/main.py:133]: The /tmp only gets cleaned on restarts?

File: lib/modelscan_api/bailo_modelscan_api/main.py [hunk 3]
```diff
@@ -101,6 +130,7 @@ def scan_file(in_file: UploadFile, background_tasks: BackgroundTasks) -> dict[st
...
             logger.info("Cleaning up downloaded file.")
             background_tasks.add_task(Path.unlink, pathlib_path, missing_ok=True)
         except UnboundLocalError:
```
Comment (JR40159) [inline lib/modelscan_api/bailo_modelscan_api/main.py:136]: When logging the exception could you include info about which file has caused the error?

File: lib/modelscan_api/bailo_modelscan_api/main.py [hunk 4]
```diff
@@ -58,33 +77,43 @@ def scan_file(in_file: UploadFile, background_tasks: BackgroundTasks) -> dict[st
...
+                        out_file.write(content)
+            except OSError as exception:
+                logger.exception("Failed writing the file to the disk.")
+                raise HTTPException(
+                    status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                    detail=f"An error occurred while trying to write the uploaded file to the disk: {exception}",
+                ) from exception
+
+            # Scan the uploaded file.
+            result = modelscan.scan(pathlib_path)
```
Comment (JR40159) [inline lib/modelscan_api/bailo_modelscan_api/main.py:113]: An info log either side of this might be useful

**Review by JR40159 — CHANGES_REQUESTED** [review]

File: lib/modelscan_api/bailo_modelscan_api/main.py (hunk 2 above)
Comment (PE39806) [reply lib/modelscan_api/bailo_modelscan_api/main.py:133]: Python's TemporaryDirectory creates a subdir of `/tmp` (e.g. `/tmp/zbc123xyz`) which is automatically cleaned up periodically https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory.cleanup

File: lib/modelscan_api/bailo_modelscan_api/main.py (hunk 3 above)
Comment (PE39806) [reply lib/modelscan_api/bailo_modelscan_api/main.py:136]: `logger.exception` automatically includes the raised exception details

**Review by JR40159 — APPROVED** [review]

**Pull request merged**